pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -1227,3 +1230,40 @@ type ScalewaySDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// IonosSDConfig configurations allow retrieving scrape targets from Ionos resources.
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ionos_sd_config
+type IonosSDConfig struct {
```

Comment:
(nit) Could you keep a general order of 
1. Fields related to the SD (DatacenterId, port etc)
2. Authorization (OAuth2.0, BasicAuth etc)
3. HTTP related